riscv/decode.h

```diff
@@ -1542,28 +1553,30 @@ for (reg_t i = 0; i < vlmax && P.VU.vl != 0; ++i) { \
   } \
   P.VU.vstart = 0;
 
-#define VI_LD(stride, offset, ld_width, elt_byte) \
+#define VI_LD(stride, offset, ld_width, elt_byte, is_seg) \
   VI_CHECK_SXX; \
-  VI_LD_COMMON(stride, offset, ld_width, elt_byte)
+  VI_LD_COMMON(stride, offset, ld_width, elt_byte, is_seg)
 
-#define VI_LD_INDEX(stride, offset, ld_width, elt_byte) \
+#define VI_LD_INDEX(stride, offset, ld_width, elt_byte, is_seg) \
   VI_CHECK_LD_INDEX; \
-  VI_LD_COMMON(stride, offset, ld_width, elt_byte) \
+  VI_LD_COMMON(stride, offset, ld_width, elt_byte, is_seg) \
   if (nf >= 2) \
     require(!is_overlapped(vd, nf, insn.rs2(), 1));
 
-#define VI_ST(stride, offset, st_width, elt_byte) \
+#define VI_ST(stride, offset, st_width, elt_byte, is_seg) \
   VI_CHECK_STORE_SXX; \
   VI_ST_COMMON(stride, offset, st_width, elt_byte) \
 
-#define VI_ST_INDEX(stride, offset, st_width, elt_byte) \
+#define VI_ST_INDEX(stride, offset, st_width, elt_byte, is_seg) \
   VI_CHECK_ST_INDEX; \
   VI_ST_COMMON(stride, offset, st_width, elt_byte) \
 
-#define VI_LDST_FF(itype, tsew) \
+#define VI_LDST_FF(itype, tsew, is_seg) \
   require(p->VU.vsew >= e##tsew && p->VU.vsew <= e64); \
-  const reg_t nf = insn.v_nf() + 1; \
+  const reg_t nf = is_seg ? insn.v_nf() + 1 : 1; \
   require((nf * P.VU.vlmul) <= (NVPR / 4)); \
+  if (!is_seg) \
+    require(nf == 1); \
```

Comment:
Same thing here

Reply:
as mentioned above, it is wrong and will be fix in next PR